src/db/inout/importcsv.cc

```diff
@@ -707,36 +711,45 @@ static std::pair<db_a_elmnt_t, persist::asset_operation>
     }
 
     auto ename = cm.get(row_i, "name");
-    if (ename.empty ())
-        bios_throw("request-param-bad", "name", TRANSLATE_ME ("<empty>"), TRANSLATE_ME ("<unique, non empty value>"));
+    if (ename.empty ()) {
+        std::string received = TRANSLATE_ME ("<empty>");
+        std::string expected = TRANSLATE_ME ("<unique, non empty value>");
+        bios_throw("request-param-bad", "name", received.c_str (), expected.c_str ());
+    }
     std::string iname;
     int rv = DBAssets::extname_to_asset_name (ename, iname);
     log_debug ("name = '%s/%s'", ename.c_str(), iname.c_str());
     if (rv == -2) {
-        bios_throw("internal-error", TRANSLATE_ME ("Database failure"));
+        std::string err = TRANSLATE_ME ("Database failure");
+        bios_throw("internal-error", err.c_str ());
     }
     unused_columns.erase("name");
 
     auto type = cm.get_strip(row_i, "type");
     log_debug ("type = '%s'", type.c_str());
     if ( TYPES.find(type) == TYPES.end() ) {
-        bios_throw("request-param-bad", "type", type.empty() ? TRANSLATE_ME ("<empty>") : JSONIFY (type.c_str()), JSONIFY(utils::join_keys_map(TYPES, ", ").c_str()));
+        std::string received = type.empty() ? TRANSLATE_ME ("<empty>") : JSONIFY (type.c_str());
```

Comment:
I guess you really don't need to get char* of type to pass it to jsonify where it should construct a std::string